src/main/java/uk/sky/cqlmigrate/CqlFileParser.java

```diff
@@ -22,37 +22,139 @@ private CqlFileParser() {}
     private static final char CQL_STATEMENT_STRING_DELIMITER = '\'';
     private static final String CQL_STATEMENT_TERMINATOR = ";";
+    private static final String CQL_COMMENT_DOUBLE_HYPEN = "--"; //Double hypen
+    private static final String CQL_MULTI_LINE_COMMENT_OPEN = "/*"; //Forward slash asterisk
+    private static final String CQL_MULTI_LINE_COMMENT_CLOSE = "*/"; //Asterisk forward slash
+    private static final Pattern CQL_MULTI_LINE_COMMENT_PATTERN = Pattern.compile("/\\*.*?\\*/", Pattern.DOTALL);
+    private static final Pattern EOL = Pattern.compile(".*\\R|.+\\z");
+    private static final String EMPTY_STR = "";
 
     static List<String> getCqlStatementsFrom(Path cqlPath) {
-        String cqlFileAsString;
+        final LineProcessor processor = new LineProcessor();
+        String original;
 
-        try (BufferedReader cqlReader = Files.newBufferedReader(cqlPath, Charsets.UTF_8)) {
-            StringBuilder stringBuilder = new StringBuilder();
-            cqlReader.lines().forEach(stringBuilder::append);
-            cqlFileAsString = stringBuilder.toString();
+        try (final Scanner in = new Scanner(cqlPath, "UTF-8")) {
+            while ((original = in.findWithinHorizon(EOL, 0)) != null) {
+                processor.process(original);
+            }
         } catch (IOException e) {
-            LOGGER.error("Failed to execute cql script {}: {}", cqlPath.getFileName(), e.getMessage());
+            LOGGER.error("Failed to process cql script {}: {}", cqlPath.getFileName(), e.getMessage());
             throw Throwables.propagate(e);
         }
 
-        checkState(cqlFileAsString.endsWith(CQL_STATEMENT_TERMINATOR), "had a non-terminated cql line: %s", cqlFileAsString);
-        return splitByStatementTerminator(cqlFileAsString);
+        processor.check();
+
+        return Arrays.asList(processor.getResult());
     }
 
-    private static List<String> splitByStatementTerminator(String cqlStatements) {
-        List<String> statementList = new ArrayList<>();
-        String candidateStatement = "";
-
-        for (String statementFragment : cqlStatements.split(CQL_STATEMENT_TERMINATOR)) {
-            candidateStatement += statementFragment;
-            // A semicolon preceded by an odd number of single quotes must be within a string, and therefore is not a statement terminator
-            if (CharMatcher.is(CQL_STATEMENT_STRING_DELIMITER).countIn(candidateStatement) % 2 == 0) {
-                statementList.add(candidateStatement);
-                candidateStatement = "";
-            } else {
-                candidateStatement += CQL_STATEMENT_TERMINATOR;
+    static class LineProcessor {
+        private static final byte INIT = 0;
+        private static final byte FIND_EOS = 1;
+        private static final byte IS_MULTI_LINE_COMMENT = 2;
+        private static final byte IS_OPEN_STMT = 3;
+        private static final byte IS_OPEN_VALUE_EXP = 4;
+        private static final byte IS_CLOSE_STMT = 5;
```

Comment:
I'm wondering if a `LineProcessorState` enum might be better to use rather then constants.